doc/sphinx/source/users-guide/locality/compilingAndExecutingMultiLocalePrograms.rst

```diff
@@ -0,0 +1,75 @@
+:title: Users Guide: Compiling and Executing Multi-Locale Programs
+
+Compiling and Executing Multi-Locale Programs
+=============================================
+
+Selecting a Communication Mechanism
+-----------------------------------
+
+All multi-locale Chapel programs are implemented using runtime
+communication routines which transparently move data and tasks between
+the locales on which the program is running.  The choice of
+communication runtime is made via the ``--comm`` compiler flag (or
+equivalently, the ``$CHPL_COMM`` environment variable).  Portable
+options for communication include:
+
+* ``none``: specifies that no communication should be performed,
+  restricting the program's execution to a single locale.
+
+* ``gasnet``: specifies that communication should be implemented using
+  the open-source GASNet library developed at Berkeley.
+
+When using the Chapel module on Cray systems, a third option is also
+available:
+
+* ``ugni``: specifies that communication should be implemented using
+  Cray's uGNI interface.
+
+In order to use any of these options, the Chapel runtime libraries
+must have been built with the specified communication layer enabled.
```

Comment:
I'm going to put this in the 'second pass' category for now.  I struggled to avoid the term 'layer' in this first pass, and really couldn't come up with a phrasing that felt at all natural ("implementation" didn't work in all the cases of 'layer' here as well as it did in the man pages).  I'd like to kick this around more as a team -- should we embrace the term layer or find a synonym -- option or ...?